src/expr/src/expr/expr_binary_nonnull.rs

```diff
@@ -383,6 +386,40 @@ fn build_at_time_zone_expr(
     Ok(expr)
 }
 
+fn build_date_trunc_expr(
+    ret: DataType,
+    l: BoxedExpression,
+    r: BoxedExpression,
+) -> Result<BoxedExpression> {
+    let expr: BoxedExpression = match r.return_type() {
+        DataType::Timestamp => Box::new(BinaryExpression::<
+            Utf8Array,
+            NaiveDateTimeArray,
+            NaiveDateTimeArray,
+            _,
+        >::new(l, r, ret, date_trunc_timestamp)),
+        DataType::Timestampz => Box::new(BinaryExpression::<
+            Utf8Array,
+            NaiveDateTimeArray,
+            NaiveDateTimeArray,
```

Comment:
`Timestampz` is represented as an `i64`, and its interpretation of year/../minute depends on timezone. So the tenary truncation operation would be a chained operation of the following 3:
* timestampz AT TIME ZONE zone -> timestamp
* truncate(field, timestamp) -> timestamp
* timestamp AT TIME ZONE zone -> timestampz